tests/dataloader/test_graph_loader.py

```diff
@@ -9,22 +9,32 @@
     "method, kwargs",
     [
         ("knn", {"k": 2}),
+        ("knn", {"k": 2, "percentile": 50}),
+        ("knn", {"k": 2, "max_distance": 50}),
         ("radius", {"max_distance": 50}),
         ("expansion", {"max_distance": 50}),
-        ("expansion", {"percentile": 50}),
     ],
 )
-def test_build_graph(method, kwargs):
+@pytest.mark.parametrize("self_loops", [True, False])
+def test_build_graph(method, self_loops, kwargs):
     gdf = sdata["blobs_circles"]
     build_graph(gdf=gdf, method=method, **kwargs)
 
 
+# invalid choices
 @pytest.mark.parametrize(
     "method, kwargs",
-    [("knn", {"k": 2}), ("radius", {"max_distance": 50}), ("expansion", {"max_distance": 50}), ("invalid")],
+    [
+        ("knn", {"k": None}),
+        ("knn", {"k": 2, "percentile": 50, "max_distance": 50}),
+        ("radius", {"k": 50}),
+        ("radius", {"max_distance": 50, "percentile": 50}),
+        ("expansion", {"k": 50}),
+        ("expansion", {"percentile": 50}),
+        ("invalid", {}),
+    ],
 )
 def test_build_graph_invalid_arguments(method, kwargs):
     gdf = sdata["blobs_circles"]
-    if method == "invalid":
-        with pytest.raises(AssertionError):
-            build_graph(gdf=gdf, method=method, **kwargs)
+    with pytest.raises(AssertionError):
+        build_graph(gdf=gdf, method=method, **kwargs)
```

Comment:
@EliHei2 the tests were not meaningful before. Now I am testing the various combinations of parameters that are invalid.